be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/WitnessValidator.scala

```diff
@@ -1,15 +1,43 @@
 package ch.epfl.pop.pubsub.graph.validators
 
+import akka.pattern.AskableActorRef
 import ch.epfl.pop.model.network.JsonRpcRequest
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.network.method.message.data.witness.WitnessMessage
+import ch.epfl.pop.model.objects.{Channel, Hash, PublicKey, Signature}
+import ch.epfl.pop.pubsub.graph.validators.MessageValidator._
 import ch.epfl.pop.pubsub.graph.{GraphMessage, PipelineError}
+import ch.epfl.pop.storage.DbActor
 
+//Similarly to the handlers, we create a WitnessValidator object which creates a WitnessValidator class instance.
+//The defaults dbActorRef is used in the object, but the class can now be mocked with a custom dbActorRef for testing purpose
+object WitnessValidator extends MessageDataContentValidator {
+  val witnessValidator = new WitnessValidator(DbActor.getInstance)
```

Comment:
```suggestion
  lazy val witnessValidator = new WitnessValidator(DbActor.getInstance)
```
for consistency with `WitnessHandler`.